pkg/processor/runtime/golang/runtime.go

```diff
@@ -119,17 +151,58 @@ func (g *golang) callEntrypoint(event nuclio.Event, functionLogger logger.Logger
 		}
 	}()
 
-	// before we call, save timestamp
-	startTime := time.Now()
+	ctx, cancel := context.WithCancel(context.Background())
+	g.cancelEventHandlingChan <- cancel
```

Comment:
Note for future async processing in Go - The channel is of size 1, so this is now blocking for every event.

Reply:
@TomerShor not really. The size is 1(not 0), which means that if channel is empty now, then write operation is not blocking. With current implementation(sync) it's not possible to have more than one event at the same time. Actually, jumping in the future, I have an idea of increasing the buffer size to maxNum, so that we can reuse this in the future as something to control gorotines per worker.